### ISSUE TITLE
[backend-error-redesign] "context canceled" issues due to timeouts

### ISSUE THREAD

**Issue by seqizz** [restic/restic#4970]
Output of `restic version`
--------------------------
Built from master on 29th of July:
```
restic 0.17.0-dev (compiled manually) compiled with go1.22.0 on linux/amd64
```

What backend/service did you use to store the repository?
---------------------------------------------------------
s3 (served via standalone minio deployment)

Problem description / Steps to reproduce
----------------------------------------

Getting "context canceled" while doing prune operations lately. 

Expected behavior
-----------------

Operations should have a chance to finish, even on extremely slow endpoints.

Actual behavior
---------------

Depending on the IO status on the hardware (which is a busy one with multiple IO-heavy processes are running in parallel, albeit sometimes slow), some operations gets canceled. E.g.

```
List(data) returned error, retrying after 1s: Get "http://myhost:8888/mybucket/?continuation-token=ZGF0YS83Ni83NmQ4OTg2MzUzOGMxMjMxOTU2MDI2Mzg2M2M4NmM0ZTM3NzNjMjU2Y2I2MjBiZWJjYzVhZDQzY2E3ZWQ0NjIwW21pbmlvX2NhY2hlOnYyLGlkOmFiZjQ5YjFkLTVjYTItNDY3MS1hNzM4LWIyMjFkZDIwNTMzOSxwOjAsczowXQ%3D%3D&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix=data%2F": context canceled
```

Do you have any idea what may have caused this?
-----------------------------------------------

A while ago I asked the question [on the forum](https://forum.restic.net/t/context-canceled-errors-on-current-master/7630) and turned out changing [one value](https://github.com/restic/restic/commit/a213b5f695bd547ef5532868552571c3bfbb0ce6) was solving the issue (or disabling the backend-error-redesign feature flag).

Wanted to open this as an issue to make it traceable, since this is a blocking thing on slow endpoints. Plus the flag has been enabled as _beta_  on release version also.

Even though the endpoint is a lot slower than expected, I don't think restic should require a "minimum speed" from endpoints. Maybe the timeout can be configured with a special argument (sorry if there is already one, I could not find digging the code).

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Still the best! :rocket:

**Comment by fkusche** [comment]
I'm unsure whether I should open a new issue, or if I shoud add this just here. Give me a note, if you want a new issue.

I just encountered this problem with `restic 0.18.1 compiled with go1.25.1 on linux/amd64`.

The command line was: `/usr/local/bin/restic backup --compression max --limit-upload 3000 --json -v --no-scan --tag=win11-vm  /var/lib/libvirt`

The process was started at 21:02 in the evening and exited on 8:36 on the next day

stdout:
```
{"message_type":"verbose_status","action":"unchanged","item":"/var/lib/libvirt/boot/","duration":0,"data_size":0,"data_size_in_repo":0,"metadata_size":0,"metadata_size_in_repo":0,"total_files":0}
{"message_type":"verbose_status","action":"modified","item":"/var/lib/libvirt/qemu/checkpoint/","duration":0.008714073,"data_size":0,"data_size_in_repo":0,"metadata_size":0,"metadata_size_in_repo":0,"total_files":0}
...
{"message_type":"verbose_status","action":"modified","item":"/var/lib/libvirt/qemu/","duration":0.063061196,"data_size":0,"data_size_in_repo":0,"metadata_size":2296,"metadata_size_in_repo":441,"total_files":0}
{"message_type":"status","seconds_elapsed":60,"percent_done":0,"files_done":3,"bytes_done":462607238,"current_files":["/var/lib/libvirt/images/win11.qcow2"]}
{"message_type":"status","seconds_elapsed":120,"percent_done":0,"files_done":3,"bytes_done":740803254,"current_files":["/var/lib/libvirt/images/win11.qcow2"]}
{"message_type":"status","seconds_elapsed":180,"percent_done":0,"files_done":3,"bytes_done":931344310,"current_files":["/var/lib/libvirt/images/win11.qcow2"]}
...
{"message_type":"status","seconds_elapsed":4860,"percent_done":0,"files_done":3,"bytes_done":28909484074,"current_files":["/var/lib/libvirt/images/win11.qcow2"]}
```
(note that `seconds_elapsed` ends at 4860. So it seems to have stalled at about 22:24 and hung for a bit more than 10 hours.)

stderr:
```
failed to refresh stale lock: context canceled
{"message_type":"exit_error","code":1,"message":"Fatal: unable to save snapshot: client.PutObject: Access Denied"}
```

**Comment by seqizz** [comment]
The error message indicates you're getting an access denied error, so I would say this is probably an issue on the storage endpoint-side. I'd suggest you to ask on [forums](https://forum.restic.net/) first.

**Comment by fkusche** [comment]
But why does it hang for 10 hours then?
